content/src/content/jcr_root/apps/core/wcm/sandbox/components/carousel/v1/carousel/clientlibs/site/js/carousel.js

```diff
@@ -244,15 +308,84 @@
             window.scrollTo(x, y);
         }
 
+        /**
+         * Retrieves the next active index, with looping
+         *
+         * @private
+         */
+        function getNextIndex() {
+            return that._active === (that._elements["item"].length - 1) ? 0 : that._active + 1;
+        }
+
+        /**
+         * Retrieves the previous active index, with looping
+         *
+         * @private
+         */
+        function getPreviousIndex() {
+            return that._active === 0 ? (that._elements["item"].length - 1) : that._active - 1;
+        }
+
         /**
          * Navigates to the item at the provided index
          *
          * @private
          * @param {Number} index The index of the item to navigate to
          */
         function navigate(index) {
+            if (index < 0 || index > (that._elements["item"].length - 1)) {
+                return;
+            }
+
             that._active = index;
             refreshActive();
+
+            // reset the auto transition interval following navigation, if not already hovering the carousel
+            if (that._elements.self.parentElement.querySelector(":hover") !== that._elements.self) {
+                resetAutoInterval();
+            }
+        }
+
+        /**
+         * Navigates to the item at the provided index and ensures the active indicator gains focus
+         *
+         * @private
+         * @param {Number} index The index of the item to navigate to
+         */
+        function navigateAndFocusIndicator(index) {
+            navigate(index);
+            focusWithoutScroll(that._elements["indicator"][index]);
+        }
+
+        /**
+         * Starts/resets automatic slide transition interval
+         *
+         * @private
+         */
+        function resetAutoInterval() {
+            if (!that._properties.auto) {
+                return;
+            }
+            clearAutoInterval();
+            that._autoInterval = window.setInterval(function() {
+                var indicators = that._elements["indicators"];
+                if (indicators !== document.activeElement && indicators.contains(document.activeElement)) {
```

Comment:
Shouldn't it be the inverted expression? If one indicator has focus, we navigate to the next slide and focus?

Reply:
that's what we're doing here

Reply:
I was confused with the first term `indicators !== document.activeElement`. Sorry.